redirects.js

```diff
@@ -19,6 +20,37 @@ const setup = (app) => {
   app.route(["/narratives/ncov", "/narratives/ncov/sit-rep"])
     .get((req, res) => res.redirect('/#ncov'));
 
+  /*
+   * Redirect to translations of narratives if the client has
+   * set language preference and the translation is available
+   *
+   * Currently only used for nCoV narratives and requires the
+   * language code to be the second to last part of the url.
+   */
+  app.route("/narratives/ncov/sit-rep/*")
+    .get(async (req, res, next) => {
+      const {source, prefixParts} = helpers.splitPrefixIntoParts(req.url);
+      const availableNarratives = await source.availableNarratives();
+      const availableLanguages = new Set(availableNarratives.map((narrative) => {
+        if (narrative.startsWith('ncov/sit-rep/')) {
+          const narrativeParts = narrative.split("/");
+          const language = narrativeParts[narrativeParts.length - 2];
+          if (language !== 'sit-rep') return language;
+        }
+        return null;
```

Comment:
This is trying to use a map as both a map and a filter, but will include `null` in the final `Set`.